ivy/stateful/module.py

```diff
@@ -549,7 +568,11 @@ def build(
         else:
             dtype = ivy.default_dtype(dtype=self._dtype, as_native=True)
 
-        kwargs["dtype"] = dtype
+        # why are we adding this kwarg in user-defined build ?
+        # it results in the error while doing `from_haiku_module` if haiku's forward
+        # therefore leaving it commented out
+        # kwargs["dtype"] = dtype
+
```

Comment:
not sure if breaks anything if added back, probably good to have it mind

Reply:
let's leave the comment there, it did break something or the other during transpilation, that's why I commented it.